src/Workspaces/Core/Portable/CodeFixesAndRefactorings/FixAllProviderInfo.cs

```diff
@@ -119,5 +145,18 @@ private class SuppressionFixerFixAllProviderInfo : FixAllProviderInfo
             public override bool CanBeFixed(Diagnostic diagnostic)
                 => _canBeSuppressedOrUnsuppressed(diagnostic);
         }
+
+        private class CodeRefactoringFixAllProviderInfo : FixAllProviderInfo
+        {
+            public CodeRefactoringFixAllProviderInfo(
+                IFixAllProvider fixAllProvider,
+                ImmutableArray<FixAllScope> supportedScopes)
+                : base(fixAllProvider, supportedScopes)
+            {
+            }
+
+            public override bool CanBeFixed(Diagnostic diagnostic)
+                => throw ExceptionUtilities.Unreachable;
```

Comment:
definitely smelly :)  what is going on here?

Reply:
`FixAllProviderInfo` has this abstract method, which is only expected to be called for the info created for code fixes: https://github.com/dotnet/roslyn/blob/dbedfa27696dba71d3f5cbe8a956bfa30cfacb21/src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/FixAllProviderInfo.cs#L87